### ISSUE TITLE
'getpwuid(): uid not found: 1001'

### ISSUE THREAD

**Issue by copyandrun** [mesonbuild/meson#11662]
**Describe the bug**
I am install dbus-python and this package require the meson. the install will raise an exception with 'getpwuid(): uid not found: 1001'

**To Reproduce**
Use an docker to install the dbus-python with the command.
sudo pip3 install dbus-python==1.3.2
```
INFO: autodetecting backend as ninja
      INFO: calculating backend command to run: /tmp/pip-build-env-v1swhhzx/overlay/bin/ninja
      Traceback (most recent call last):
        File "/tmp/pip-build-env-v1swhhzx/overlay/lib/python3.9/site-packages/mesonbuild/mesonmain.py", line 194, in run
          return options.run_func(options)
        File "/tmp/pip-build-env-v1swhhzx/overlay/lib/python3.9/site-packages/mesonbuild/minstall.py", line 823, in run
          if not rebuild_all(opts.wd, backend):
        File "/tmp/pip-build-env-v1swhhzx/overlay/lib/python3.9/site-packages/mesonbuild/minstall.py", line 803, in rebuild_all
          env, preexec_fn = drop_privileges()
        File "/tmp/pip-build-env-v1swhhzx/overlay/lib/python3.9/site-packages/mesonbuild/minstall.py", line 779, in drop_privileges
          homedir = pwd.getpwuid(int(orig_uid)).pw_dir
      KeyError: 'getpwuid(): uid not found: 1001'
```

**Expected behavior**
install successful

**system parameters**
python 3.9.9 with meson 1.1.0

I check it successful yesterday. I guess it should be the new issue in 1.1.0 since it is a new release version.
I found it with https://github.com/mesonbuild/meson/commit/ecb32bf457ed27b75c9b4386ca1acfbd4b63869e

**Comment by copyandrun** [comment]
I don't think it is a docker issue. The docker image https://repo.openeuler.org/openEuler-23.03/docker_img/x86_64/

**Comment by dnicolodi** [comment]
The issue is the `chroot`. Most likely there is no `/etc/passwd` or just a stub in the chroot. This explains why the `getpwuid()` call fails. However, in this case dropping privileges is not the right thing to do. I think the fix in #11667 should cover this case too.

**Comment by eli-schwartz** [comment]
So you are running sudo, which works fine, and then running a container command (chroot) which is ***not*** docker, and that doesn't wipe environment variables. In such a situation I'd probably use systemd-nspawn to be honest.

Thanks for clarifying the root cause, I was baffled at how this tied into docker and the initial attempt I made at solving this included comments which blamed docker which would have been fairly confusing.

I agree that the fix in my PR should handle this too, and I will update it.
